hamcrest/Hamcrest/Arrays/SeriesMatchingOnce.php

```diff
@@ -31,7 +31,7 @@ public function matches($item)
     public function isFinished()
     {
         if (!empty($this->_elementMatchers)) {
-            $nextMatcher = current($this->_elementMatchers);
+            $nextMatcher = reset($this->_elementMatchers);
```

Comment:
current() will return "false", if the end of a array way reached. That was the case with the failing tests, there were 3 matchers in the array, but the internal index of that array was behind the last element.